File: projecteuler/0 - 100/001/multiplesOf3or5.py
from sys import argv


# Non-functional Calculation
def multiplesOf3or5(n):
    sum = 0
    for i in range(0, int(n)):
        if(i % 3 == 0 or i % 5 ==0):
            sum += i
    return sum


# Functional Calculation
def multiplesOf3or5Functional(n):
    return sum(filter(lambda x: x % 3 == 0 or x % 5 == 0, range(0, int(n))))


# Main
if __name__ == "__main__":
    try:
        print(f"\nSum of multiples of 3 or 5 below {argv[1]}")
        print(f"Non-functional: {multiplesOf3or5(argv[1])}")
        print(f"Functional: {multiplesOf3or5Functional(argv[1])}\n")
    except Exception as error:
        print(f"Error occured: {error}")

File: projecteuler/0 - 100/001/multiplesOf3or5.scala
object multiplesOf3or5 {
    def findSum(n: String): Int = {
        // Define the filter used
        def is3or5(i: Int): Boolean = i % 3 == 0 || i % 5 == 0
        // Return the sum of filter applied to list
        return List.range(0, n.toInt).filter(is3or5).sum
    }
    def main(args: Array[String]) {
        val n = args(0)
        println(s"\nSum of multiples of 3 or 5 below $n:")
        println(findSum(n))
    }
}